uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/models/instances/absence-period-instance_test.js

```diff
@@ -0,0 +1,81 @@
+define([
+  'mocks/data/absence-period-data',
+  'leave-absences/shared/models/instances/absence-period-instance',
+], function (mockData) {
+  'use strict'
+
+  describe('AbsencePeriodInstance', function () {
+    var AbsencePeriodInstance, ModelInstance;
+
+    beforeEach(module('leave-absences.models.instances'));
+
+    beforeEach(inject(function (_AbsencePeriodInstance_, _ModelInstance_) {
+      AbsencePeriodInstance = _AbsencePeriodInstance_;
+      ModelInstance = _ModelInstance_;
+    }));
+
+    it('inherits from ModelInstance', function () {
+      expect(_.functions(AbsencePeriodInstance)).toEqual(jasmine.arrayContaining(_.functions(ModelInstance)));
+    });
+
+    describe('init()', function () {
```

Comment:
Better structure
```js
describe('init()', function () {
  describe('basic properties', function () {
    beforeEach(function () {
      instance = AbsencePeriodInstance.init(attributes, true);
    });

    it('has expected data', function () {

    });
  });

  describe('`current` property', function () {
    describe('when the absence period is current', function () {
      beforeEach(function () {
        instance = // init with current absence period
      });

      it('has `current` set to `true`', function () {

      });
    });

    describe('when the absence period is past', function () {
      beforeEach(function () {
        instance = // init with current absence period
      });

      it('has `current` set to `false`', function () {

      });
    });
  });
});
```
The goal is to have the test suite to almost read like plain english if you only read the `describe` and `it` labels